### PR TITLE
scrcpy: new port

### PR DESCRIPTION
**Pull request by MarcelBochtler**
Add scrcpy a tool to display and control Android devices.

Resolves: https://trac.macports.org/ticket/60440

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.7 19H2
Xcode 12.4 12D4e

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by tobypeterson** [comment]
Fetching resources from the network during configure is not ok. You should add that as an additional distfile (i.e. with `distfiles-append`).

**Comment by MarcelBochtler** [comment]
@tobypeterson Thanks for the hint. I replaced the fetching during configure with `distfiles-append` as you suggested.

File: multimedia/scrcpy/Portfile [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+name                scrcpy
+version             1.17
```
Comment (judaew) [inline multimedia/scrcpy/Portfile:7]: The name and version will be taken from `github.setup`:

```diff
PortSystem          1.0
PortGroup           github 1.0

- name scrcpy
- version 1.17
+ github.setup        Genymobile scrcpy 1.17 v
+ github.tarball_from releases
```

File: multimedia/scrcpy/Portfile [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                    size    266684 \
+                    scrcpy-server-v${version} \
+                    rmd160  59bc2a451016f5c5f11a22643a441a19e3a78b9b \
+                    sha256  11b5ad2d1bc9b9730fb7254a78efd71a8ff46b1938ff468e47a21b653a1b6725 \
+                    size    34930
+
+github.tarball_from releases
+
+depends_fetch-append \
+                    port:wget
```
Comment (judaew) [inline multimedia/scrcpy/Portfile:37]: Don't forget to remove this depend.

File: multimedia/scrcpy/Portfile [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+categories          multimedia
+platforms           darwin
+license             Apache-2
+maintainers         {bochtler.io:macports @MarcelBochtler} \
+                    openmaintainer
+description         Display and control your Android device
+long_description    This application provides display and control of Android \
+                    devices connected on USB (or over TCP/IP). It does not \
+                    require any root access.
+homepage            https://github.com/Genymobile/scrcpy
```
Comment (mf2k) [inline multimedia/scrcpy/Portfile:17]: The `homepage` is already set to this by `github.setup`. So this redundant line should be deleted.

File: multimedia/scrcpy/Portfile [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+github.setup        Genymobile scrcpy 1.17 v
+github.tarball_from releases
```
Comment (mf2k) [inline multimedia/scrcpy/Portfile:20]: Move these two lines to just above the `revision` line.

File: multimedia/scrcpy/Portfile [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                    scrcpy-server-v${version} \
+                    rmd160  59bc2a451016f5c5f11a22643a441a19e3a78b9b \
+                    sha256  11b5ad2d1bc9b9730fb7254a78efd71a8ff46b1938ff468e47a21b653a1b6725 \
+                    size    34930
+
+depends_build-append \
+                    port:pkgconfig \
+                    port:meson
+
+depends_lib-append  port:ffmpeg \
```
Comment (mf2k) [inline multimedia/scrcpy/Portfile:37]: Can ffmpeg-devel also be used to satisfy this dependency? If so this is normally written as:

`path:lib/libavcodec.dylib:ffmpeg
`

File: multimedia/scrcpy/Portfile [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                    sha256  b3f5490970dbef6f50725b4d46ba07baed9c7eb19696e71029257a01044c6664 \
+                    size    266684 \
+                    scrcpy-server-v${version} \
+                    rmd160  59bc2a451016f5c5f11a22643a441a19e3a78b9b \
+                    sha256  11b5ad2d1bc9b9730fb7254a78efd71a8ff46b1938ff468e47a21b653a1b6725 \
+                    size    34930
+
+depends_build-append \
+                    port:pkgconfig \
+                    port:meson
```
Comment (mf2k) [inline multimedia/scrcpy/Portfile:35]: Nitpick: Please sort these alphabetically.

**Review by mf2k — CHANGES_REQUESTED** [review]

File: multimedia/scrcpy/Portfile (hunk 5 above)
Comment (MarcelBochtler) [reply multimedia/scrcpy/Portfile:37]: Just checked. This works as well.

File: multimedia/scrcpy/Portfile [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+platforms           darwin
+license             Apache-2
+maintainers         {bochtler.io:macports @MarcelBochtler} \
+                    openmaintainer
+description         Display and control your Android device
+long_description    This application provides display and control of Android \
+                    devices connected on USB (or over TCP/IP). It does not \
+                    require any root access.
+
+distfiles-append scrcpy-server-v${version}
```
Comment (herbygillot) [inline multimedia/scrcpy/Portfile:20]: Minor nitpick, but for this line, can we match the alignment of the other lines in this file?

File: multimedia/scrcpy/Portfile [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+depends_build-append \
+                    port:meson \
+                    port:pkgconfig
+
+depends_lib-append  path:lib/libavcodec.dylib:ffmpeg \
+                    port:libsdl2
+
+configure.cmd       meson x --buildtype release --strip -Db_lto=true -Dprebuilt_server=${distpath}/scrcpy-server-v${version}
+
+build.cmd           ninja -Cx
```
Comment (mf2k) [inline multimedia/scrcpy/Portfile:40]: There is a meson portgroup. Did you investigate using it? It would be preferred to use it if you can.

**Review by mf2k — CHANGES_REQUESTED** [review]

File: multimedia/scrcpy/Portfile [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+destroot {
+    xinstall -m 755 -d ${destroot}${prefix}/share/${name}
+    xinstall -m 644 ${worksrcpath}/x/server/scrcpy-server ${destroot}${prefix}/share/${name}
+
+    xinstall -m 755 ${worksrcpath}/x/app/scrcpy ${destroot}${prefix}/bin
+}
+
+notes "
+${name} requires adb. It can be installed using port install android-platform-tools.
+"
```
Comment (harens) [inline multimedia/scrcpy/Portfile:51]: Seriously looking forward to this. Thanks @MarcelBochtler!

Can `android-platform-tools` be set in `depends_run`, similar to ports such as [logcat-color3](https://github.com/macports/macports-ports/tree/master/python/logcat-color3/Portfile)?

In that port they use the following, which should hopefully also be applicable here:

```tcl
depends_run-append \
                bin:adb:android-platform-tools
```

Comment (mf2k) [reply multimedia/scrcpy/Portfile:51]: Yes, that would be much better.

File: multimedia/scrcpy/Portfile (hunk 8 above)
Comment (MarcelBochtler) [reply multimedia/scrcpy/Portfile]: Done.

File: multimedia/scrcpy/Portfile (hunk 9 above)
Comment (MarcelBochtler) [reply multimedia/scrcpy/Portfile]: Done

File: multimedia/scrcpy/Portfile (hunk 7 above)
Comment (MarcelBochtler) [reply multimedia/scrcpy/Portfile]: Done

File: multimedia/scrcpy/Portfile [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+                    sha256  11b5ad2d1bc9b9730fb7254a78efd71a8ff46b1938ff468e47a21b653a1b6725 \
+                    size    34930
+
+depends_build-append \
+                    port:pkgconfig
+
+depends_lib-append  path:lib/libavcodec.dylib:ffmpeg \
+                    port:libsdl2
+
+depends_run-append  bin:adb:android-platform-tools
```
Comment (mf2k) [inline multimedia/scrcpy/Portfile:38]: There is no devel version of android-platform-tools available in Macports. So make this a port dependency. 

`port:android-platform-tools`

**Review by mf2k — CHANGES_REQUESTED** [review]

**Review by mf2k — APPROVED** [review]

**Comment by mf2k** [comment]
Any ideas why the builder is failing?

**Comment by judaew** [comment]
The problem is that distfiles come from different sources. The source code is taken from the GitHub archive, and scrcpy-server from the GitHub releases. See https://github.com/Genymobile/scrcpy/releases/tag/v1.17

Any idea how @MarcelBochtler can use both variants fetch with the github PortGroup?

File: multimedia/scrcpy/Portfile [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+license             Apache-2
+maintainers         {bochtler.io:macports @MarcelBochtler} \
+                    openmaintainer
+description         Display and control your Android device
+long_description    This application provides display and control of Android \
+                    devices connected on USB (or over TCP/IP). It does not \
+                    require any root access.
+
+distfiles-append    scrcpy-server-v${version}
+
```
Comment (judaew) [inline multimedia/scrcpy/Portfile:22]: Try, I think it should work

```diff
@@ line 7 @@
- github.tarball_from releases

@@ line 21-22 @@
+ master_sites-append https://github.com/Genymobile/${name}/releases/download/v${version}/:bootstrap
+ distfiles-append scrcpy-server-v${version}

Comment (MarcelBochtler) [reply multimedia/scrcpy/Portfile]: Thanks. That seems to work.

**Review by judaew — APPROVED** [review]

**Review by herbygillot — APPROVED** [review]

**Review by harens — APPROVED** [review]
Lgtm. Thanks @MarcelBochtler for your work!

**Pull request merged**